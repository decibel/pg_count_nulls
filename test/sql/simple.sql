\set ECHO none

\set schema public

\i test/helpers/setup.sql

\i test/core/functions.sql

--SET client_min_messages = debug;

SELECT * FROM runtests( '_null_count_test'::name );
