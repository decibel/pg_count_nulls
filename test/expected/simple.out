\set ECHO none
    # Subtest: _null_count_test.test__definition()
    ok 1 - Function null_count(anyarray) should return bigint
    ok 2 - Function null_count(anyarray) should not be strict
    ok 3 - Function null_count(anyarray) should be IMMUTABLE
    ok 4 - Function null_count(json) should return bigint
    ok 5 - Function null_count(json) should not be strict
    ok 6 - Function null_count(json) should be IMMUTABLE
    ok 7 - Function not_null_count(anyarray) should return bigint
    ok 8 - Function not_null_count(anyarray) should not be strict
    ok 9 - Function not_null_count(anyarray) should be IMMUTABLE
    ok 10 - Function not_null_count(json) should return bigint
    ok 11 - Function not_null_count(json) should not be strict
    ok 12 - Function not_null_count(json) should be IMMUTABLE
    ok 13 - Function null_count_trigger() should return trigger
    ok 14 - Function null_count_trigger() should not be strict
    ok 15 - Function null_count_trigger() should be IMMUTABLE
    ok 16 - Function not_null_count_trigger() should return trigger
    ok 17 - Function not_null_count_trigger() should not be strict
    ok 18 - Function not_null_count_trigger() should be IMMUTABLE
    1..18
ok 1 - _null_count_test.test__definition
    # Subtest: _null_count_test.test__functionality()
    ok 1 - Test null_count(a, b, c)
    ok 2 - Test null_count(a, b, c)
    ok 3 - CREATE TRIGGER "test trigger" BEFORE INSERT ON test_data FOR EACH ROW EXECUTE PROCEDURE not_null_count_trigger( NULL )
    ok 4 - Test not_null_count_trigger( NULL )
    ok 5 - DROP TRIGGER "test trigger"
    ok 6 - CREATE TRIGGER "test trigger" BEFORE INSERT ON test_data FOR EACH ROW EXECUTE PROCEDURE not_null_count_trigger(  )
    ok 7 - Test not_null_count_trigger(  )
    ok 8 - DROP TRIGGER "test trigger"
    ok 9 - CREATE TRIGGER "test trigger" BEFORE INSERT ON test_data FOR EACH ROW EXECUTE PROCEDURE null_count_trigger( NULL )
    ok 10 - Test null_count_trigger( NULL )
    ok 11 - DROP TRIGGER "test trigger"
    ok 12 - CREATE TRIGGER "test trigger" BEFORE INSERT ON test_data FOR EACH ROW EXECUTE PROCEDURE null_count_trigger(  )
    ok 13 - Test null_count_trigger(  )
    ok 14 - DROP TRIGGER "test trigger"
    ok 15 - CREATE TRIGGER "null_BEFORE_error_message" BEFORE INSERT ON test_data FOR EACH ROW EXECUTE PROCEDURE null_count_trigger(1, 'error_message')
    ok 16 - Test "null_BEFORE_error_message"
    ok 17 - DROP TRIGGER "null_BEFORE_error_message"
    ok 18 - CREATE TRIGGER "null_AFTER_error_message" AFTER INSERT ON test_data FOR EACH ROW EXECUTE PROCEDURE null_count_trigger(1, 'error_message')
    ok 19 - Test "null_AFTER_error_message"
    ok 20 - DROP TRIGGER "null_AFTER_error_message"
    ok 21 - CREATE TRIGGER "not_null_BEFORE_error_message" BEFORE INSERT ON test_data FOR EACH ROW EXECUTE PROCEDURE not_null_count_trigger(1, 'error_message')
    ok 22 - Test "not_null_BEFORE_error_message"
    ok 23 - DROP TRIGGER "not_null_BEFORE_error_message"
    ok 24 - CREATE TRIGGER "not_null_AFTER_error_message" AFTER INSERT ON test_data FOR EACH ROW EXECUTE PROCEDURE not_null_count_trigger(1, 'error_message')
    ok 25 - Test "not_null_AFTER_error_message"
    ok 26 - DROP TRIGGER "not_null_AFTER_error_message"
    ok 27 - CREATE TRIGGER "null_BEFORE_" BEFORE INSERT ON test_data FOR EACH ROW EXECUTE PROCEDURE null_count_trigger(1, NULL)
    ok 28 - Test "null_BEFORE_"
    ok 29 - DROP TRIGGER "null_BEFORE_"
    ok 30 - CREATE TRIGGER "null_AFTER_" AFTER INSERT ON test_data FOR EACH ROW EXECUTE PROCEDURE null_count_trigger(1, NULL)
    ok 31 - Test "null_AFTER_"
    ok 32 - DROP TRIGGER "null_AFTER_"
    ok 33 - CREATE TRIGGER "not_null_BEFORE_" BEFORE INSERT ON test_data FOR EACH ROW EXECUTE PROCEDURE not_null_count_trigger(1, NULL)
    ok 34 - Test "not_null_BEFORE_"
    ok 35 - DROP TRIGGER "not_null_BEFORE_"
    ok 36 - CREATE TRIGGER "not_null_AFTER_" AFTER INSERT ON test_data FOR EACH ROW EXECUTE PROCEDURE not_null_count_trigger(1, NULL)
    ok 37 - Test "not_null_AFTER_"
    ok 38 - DROP TRIGGER "not_null_AFTER_"
    1..38
ok 2 - _null_count_test.test__functionality
1..2
